project2/py/knowledge_base.py

```diff
@@ -22,3 +22,21 @@ def get_meat_substitute(self):
         index = random.randint(0, upper_bound)
         chosen_sub = self.meat_substitutes[index]
         return chosen_sub
+
+    def extract_methods(self, direction):
+        methods = ''
+        direction = direction.lower().replace('.,', '')
```

Comment:
I think this text processing is best done before this step. I think you can assume the directions are lower case at this point.